bluemira/magnets/materials.py

```diff
@@ -0,0 +1,463 @@
+        """Specific heat (constant volume)"""
+        pass
+
+
+class Material(ABCMaterial):
+    """Generic Material"""
+
+    def density(self, **kwargs):  # noqa: ARG002
+        """Density"""
+        return 0
+
+    def E(self, **kwargs):  # noqa: N802, ARG002
+        """Young's module"""
+        return 1e-6
+
+    def erho(self, **kwargs):  # noqa: ARG002
+        """Electrical Resistivity"""
+        return 1e6
+
+    def cp_v(self, **kwargs):  # noqa: ARG002
+        """Specific heat (constant volume)"""
+        return 0
+
+
+class AISI_316LN(Material):  # noqa: N801
+    """
+    Stainless steel (316 LN, with extra low carbon content, aged)
+
+    Ref
+    ---
+    EFDA Material Data Compilation for Superconductor Simulation
+    P. Bauer, H. Rajainmaki, E. Salpietro
+    EFDA CSU, Garching, 04/18/07
+
+    """
+
+    def density(self, **kwargs):  # noqa: ARG002
+        """Material density [kg/m³]"""
+        return 7890.0
+
+    def E(self, T: float, **kwargs):  # noqa: N802, N803, ARG002
+        """
+        Young's module
+
+        Ref
+        ---
+        Data fits from N. Mitchell, “Finite element simulations of elasto-plastic
+        processes” Cryogenics 45 (2005) 501–515
+
+        Parameters
+        ----------
+        T:
+            Operating temperature [K]
+
+        Returns
+        -------
+            float [Pa]
+        """
+        if T > 173:
+            E = 200.4 * 1e9 - 8.1221e-2 * (T - 273) * 1e9
+        else:
+            E = 208.5 * 1e9
+        return E
+
+    def erho(self, T: float, **kwargs):  # noqa: N803
+        """
+        Electrical Resistivity
+
+        Ref
+        ---
+        Fit of data quoted by J. Davis in ITER Material Properties Handbook,
+        ITER Document No. S 74 RE 1
+
+        Parameters
+        ----------
+        T:
+            Operating temperature [K]
+
+        Return
+        ------
+            float [Ohm m]
+        """
+        return (
+                76.2063 + (0.071375 * (T - 273.15)) - (2.3109 * 1e-5 * (T - 273.15) ** 2)
+        ) * 1e-8
+
+
+class Copper100(Material):
+    def __init__(self):
+        """
+        Copper (OFHC, high purity RRR~100, annealed)
+
+        Ref
+        ---
+        EFDA Material Data Compilation for Superconductor Simulation
+        P. Bauer, H. Rajainmaki, E. Salpietro
+        EFDA CSU, Garching, 04/18/07
+
+        """
+        # cp300: Specific heat known data-point at 300 K [J/K/m³]
+        self.cp300 = 3.454e6
+        # gamma: Debye fit parameter for Cp calculation [J/K**2/kg]
+        self.gamma = 0.011
+        # beta: Grueneisen fit parameter for Cp calculation [J/K**4/kg]
+        self.beta = 0.0011
+        # RRR: Residual-resistance ratio (ratio of the resistivity of a material at
+        # room temperature and at 0 K)
+        self.RRR = 100
+
+    def density(self, **kwargs):
+        """Material density [kg/m³]"""
+        return 8960
+
+    def erho(self, T: float, B: float, **kwargs):
```

Comment:
I picked this equation to review at random (well, because it looked more complicated). I am not able to retrace its origins entirely. From the reference in the docstring I can find parts of the below from equations 8-1 and 8-7, but there are some discrepancies from what I can tell, or perhaps just some undocumented values that differ from those in the reference (e.g. rho_c in 8-1).